modules/CbChargeSOMDriver/evse_board_support/evse_board_supportImpl.cpp

```diff
@@ -0,0 +1,414 @@
+                    Everest::error::Severity::High);
+                this->raise_error(error_object);
+                this->pp_fault_reported = true;
+            }
+
+            if (this->pp_ampacity.ampacity != types::board_support_common::Ampacity::None &&
+                this->pp_fault_reported) {
+                // clear a ProximityFault error on PP state change to a valid value but only if it exists
+                this->clear_error("evse_board_support/MREC23ProximityFault");
+                this->pp_fault_reported = false;
+            }
+        }
+    } catch (std::underflow_error& e) {
+        if (!this->pp_fault_reported) {
+            EVLOG_error << e.what();
+
+            // publish a ProximityFault
+            Everest::error::Error error_object = this->error_factory->create_error(
+                "evse_board_support/MREC23ProximityFault", "", e.what(), Everest::error::Severity::High);
+            this->raise_error(error_object);
+
+            this->pp_fault_reported = true;
+        }
+    }
+
+    this->pp_observation_lock.unlock();
+
+    //EVLOG_debug << "Proximity Pilot Observation Callback finished";
+}
+
+void evse_board_supportImpl::handle_ac_set_overcurrent_limit_A(double& value) {
+    // your code for cmd ac_set_overcurrent_limit_A goes here
+}
+
+void evse_board_supportImpl::disable_cp_observation() {
+    if (this->cp_observation_enabled) {
+        EVLOG_debug << "Disabling CP observation";
+        this->cp_observation_lock.lock();
+        this->cp_observation_enabled = false;
+    } else {
+        EVLOG_debug << "Disabling CP observation (suppressed)";
+    }
+}
+
+void evse_board_supportImpl::enable_cp_observation() {
+    if (!this->cp_observation_enabled) {
+        EVLOG_debug << "Enabling CP observation";
+        this->cp_observation_enabled = true;
+        this->cp_observation_lock.unlock();
+    } else {
+        EVLOG_debug << "Enabling CP observation (suppressed)";
+    }
+}
+
+bool evse_board_supportImpl::cp_state_changed(struct cp_state_signal_side& signal_side) {
+    bool rv {false};
+
+    // CP state is only detected if the new state is different from the previous one (first condition).
+    // Additionally, to filter simple disturbances, a new state must be detected twice before notifying it (second
+    // condition). For that, we need at least two CP state measurements (third condition)
+
+    if (signal_side.previous_state != signal_side.measured_state &&
+        signal_side.current_state == signal_side.measured_state) {
+
+        // update the previous state
+        signal_side.previous_state = signal_side.current_state;
+        rv = true;
+
+    } else if (signal_side.measured_state == signal_side.previous_state &&
+               signal_side.measured_state != signal_side.current_state) {
+        EVLOG_warning << "CP state change from " << signal_side.previous_state << " to " << signal_side.current_state
+                      << " suppressed";
```

Comment:
We have the same code in CbTarragonDriver, I just copied. So I'll leave it as is for the moment since this is code which will go away sooner or later.